applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -63,21 +63,20 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientPwLineElement : public Elem
 
     void GetDofList(DofsVectorType& rElementalDofList, const ProcessInfo&) const override
     {
-        rElementalDofList = GetDofs();
+        rElementalDofList = MakeDofGetter();
     }
 
     void EquationIdVector(EquationIdVectorType& rResult, const ProcessInfo&) const override
     {
-        rResult = Geo::DofUtilities::ExtractEquationIdsFrom(GetDofs());
+        rResult = Geo::DofUtilities::ExtractEquationIdsFrom(MakeDofGetter());
     }
 
-    void Initialize(const ProcessInfo& rCurrentProcessInfo) override
+    void Initialize(const ProcessInfo&) override
     {
-        if (const std::size_t number_integration_points =
-                GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
-            mRetentionLawVector.size() != number_integration_points) {
-            mRetentionLawVector.resize(number_integration_points);
-        }
+        const std::size_t number_integration_points =
+            GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+        mRetentionLawVector.resize(number_integration_points);
```

Comment:
Since `number_of_integration_points` is only used here and the function name is very clear, I'd suggest to use a temporary:
```suggestion
        mRetentionLawVector.resize(GetGeometry().IntegrationPointsNumber(GetIntegrationMethod()));
```

Reply:
Done